src/senaite/core/patches/archetypes/referenceable.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import api
+from Products.Archetypes.config import UID_CATALOG
+
+
+def _catalogUID(self, aq, uc=None):
+    # skip indexing of temporary objects
+    if api.is_temporary(self):
+        return
+    if not uc:
+        uc = api.get_tool(UID_CATALOG)
+    url = self._getURL()
```

Comment:
Note for @xispa : `._getURL()` relies on `Archetypes.utils.getRelPath`, that returns the relative path of the object from the portal root. The result is the same as `url = "/".join(obj.getPhysicalPath()[2:])`